### ISSUE TITLE
Select/dropdown - suppress known false positive in Storybook accessibility tests

### ISSUE THREAD

**Issue by mkernohanbc** [bcgov/design-system#367]
The automated accessibility testing we run within Storybook (using the [storybook-addon-a11y](https://storybook.js.org/addons/@storybook/addon-a11y) extension) throws a 'serious violation' warning on the Select component: 

> ARIA hidden element must not be focusable or contain focusable elements

![Image](https://github.com/bcgov/design-system/assets/135075821/be9f71a9-82ac-49ae-88e0-dc7848b98405)

This is [a known false positive within RAC's Select component](https://react-spectrum.adobe.com/react-aria/Select.html#false-positives): 

> Select may trigger a [known accessibility false positive](https://github.com/adobe/react-spectrum/wiki/Known-accessibility-false-positives#picker) from automated accessibility testing tools. This is because we include a visually hidden select element alongside the Select to specifically aid with browser form autocomplete and hide it from screen readers via aria-hidden since users don't need to interact with the hidden select. We manage focus internally so that focusing this hidden select will always shift focus to the Select's trigger button instead. Automated accessibility testing tools have no way of knowing that we manage the focus in this way and thus throw this false positive.

RAC provides an attribute we can target to override and suppress this warning in storybook-addon-a11y.

**Comment by mkernohanbc** [comment]
Thinking about this a little more, probably a good idea to include a note about this false positive in the docs for the component too — it's going to pop up if users are running similar automated tests, they need to know what's going on. @ty2k what do you think?
